### ISSUE TITLE
Build on ubuntu 20.04 failed

### ISSUE THREAD

**Issue by kenken64** [Genymobile/scrcpy#1571]
he Meson build system
Version: 0.53.2
Source dir: /home/kenneth/Projects/scrcpy
Build dir: /home/kenneth/Projects/scrcpy/build_test
Build type: native build
Project name: scrcpy
Project version: 1.14
Appending CFLAGS from environment: '-march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -pipe'
Appending LDFLAGS from environment: ['-Wl,-O2', '-Wl,--sort-common', '-Wl,--as-needed', '-Wl,-z,relro', '-Wl,-z,now']
Appending CPPFLAGS from environment: '-DNDEBUG -D_FORTIFY_SOURCE=2 -O2'
Appending CFLAGS from environment: '-march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -pipe'
Appending LDFLAGS from environment: ['-Wl,-O2', '-Wl,--sort-common', '-Wl,--as-needed', '-Wl,-z,relro', '-Wl,-z,now']
Appending CPPFLAGS from environment: '-DNDEBUG -D_FORTIFY_SOURCE=2 -O2'
C compiler for the host machine: /home/kenneth/anaconda3/bin/x86_64-conda_cos6-linux-gnu-cc (gcc 7.2.0 "x86_64-conda_cos6-linux-gnu-cc (crosstool-NG fa8859cb) 7.2.0")
C linker for the host machine: /home/kenneth/anaconda3/bin/x86_64-conda_cos6-linux-gnu-cc ld.bfd 2.28.1
Host machine cpu family: x86_64
Host machine cpu: x86_64
Found pkg-config: /usr/bin/pkg-config (0.29.1)
Run-time dependency libavformat found: YES 58.29.100
Run-time dependency libavcodec found: YES 58.54.100
Run-time dependency libavutil found: YES 56.31.100
Run-time dependency sdl2 found: YES 2.0.10
Configuring config.h using configuration
Program ./scripts/build-wrapper.sh found: YES (/home/kenneth/Projects/scrcpy/server/./scripts/build-wrapper.sh)
Build targets in project: 10

Found ninja-1.10.0 at /usr/bin/ninja
ninja: Entering directory `build_test'
[1/47] Compiling C object 'app/a172ced@@scrcpy@exe/src_controller.c.o'.
FAILED: app/a172ced@@scrcpy@exe/src_controller.c.o 
/home/kenneth/anaconda3/bin/x86_64-conda_cos6-linux-gnu-cc -Iapp/a172ced@@scrcpy@exe -Iapp -I../app -I../app/src -I/usr/include/x86_64-linux-gnu -I/usr/include/SDL2 -fdiagnostics-color=always -fsanitize=address -fno-omit-frame-pointer -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -g -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -D_REENTRANT -MD -MQ 'app/a172ced@@scrcpy@exe/src_controller.c.o' -MF 'app/a172ced@@scrcpy@exe/src_controller.c.o.d' -o 'app/a172ced@@scrcpy@exe/src_controller.c.o' -c ../app/src/controller.c
In file included from ../app/src/controller.c:1:0:
../app/src/controller.h:5:10: fatal error: SDL2/SDL_mutex.h: No such file or directory
 #include <SDL2/SDL_mutex.h>
          ^~~~~~~~~~~~~~~~~~
compilation terminated.
[2/47] Compiling C object 'app/a172ced@@scrcpy@exe/src_cli.c.o'.
FAILED: app/a172ced@@scrcpy@exe/src_cli.c.o 
/home/kenneth/anaconda3/bin/x86_64-conda_cos6-linux-gnu-cc -Iapp/a172ced@@scrcpy@exe -Iapp -I../app -I../app/src -I/usr/include/x86_64-linux-gnu -I/usr/include/SDL2 -fdiagnostics-color=always -fsanitize=address -fno-omit-frame-pointer -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -g -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -D_REENTRANT -MD -MQ 'app/a172ced@@scrcpy@exe/src_cli.c.o' -MF 'app/a172ced@@scrcpy@exe/src_cli.c.o.d' -o 'app/a172ced@@scrcpy@exe/src_cli.c.o' -c ../app/src/cli.c
In file included from /home/kenneth/anaconda3/x86_64-conda_cos6-linux-gnu/sysroot/usr/include/features.h:356:0,
                 from /home/kenneth/anaconda3/x86_64-conda_cos6-linux-gnu/sysroot/usr/include/stdint.h:26,
                 from /home/kenneth/anaconda3/lib/gcc/x86_64-conda_cos6-linux-gnu/7.2.0/include/stdint.h:9,
                 from ../app/src/scrcpy.h:5,
                 from ../app/src/cli.h:7,
                 from ../app/src/cli.c:1:
/usr/include/x86_64-linux-gnu/sys/cdefs.h:492:49: error: missing binary operator before token "("
 #if __GNUC_PREREQ (4,8) || __glibc_clang_prereq (3,5)
                                                 ^
In file included from ../app/src/input_manager.h:8:0,
                 from ../app/src/scrcpy.h:9,
                 from ../app/src/cli.h:7,
                 from ../app/src/cli.c:1:
../app/src/controller.h:5:10: fatal error: SDL2/SDL_mutex.h: No such file or directory
 #include <SDL2/SDL_mutex.h>
          ^~~~~~~~~~~~~~~~~~
compilation terminated.
[3/47] Compiling C object 'app/a172ced@@scrcpy@exe/src_main.c.o'.
FAILED: app/a172ced@@scrcpy@exe/src_main.c.o 
/home/kenneth/anaconda3/bin/x86_64-conda_cos6-linux-gnu-cc -Iapp/a172ced@@scrcpy@exe -Iapp -I../app -I../app/src -I/usr/include/x86_64-linux-gnu -I/usr/include/SDL2 -fdiagnostics-color=always -fsanitize=address -fno-omit-frame-pointer -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -g -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -D_REENTRANT -MD -MQ 'app/a172ced@@scrcpy@exe/src_main.c.o' -MF 'app/a172ced@@scrcpy@exe/src_main.c.o.d' -o 'app/a172ced@@scrcpy@exe/src_main.c.o' -c ../app/src/main.c
In file included from /home/kenneth/anaconda3/x86_64-conda_cos6-linux-gnu/sysroot/usr/include/features.h:356:0,
                 from /home/kenneth/anaconda3/x86_64-conda_cos6-linux-gnu/sysroot/usr/include/stdint.h:26,
                 from /home/kenneth/anaconda3/lib/gcc/x86_64-conda_cos6-linux-gnu/7.2.0/include/stdint.h:9,
                 from ../app/src/scrcpy.h:5,
                 from ../app/src/main.c:1:
/usr/include/x86_64-linux-gnu/sys/cdefs.h:492:49: error: missing binary operator before token "("
 #if __GNUC_PREREQ (4,8) || __glibc_clang_prereq (3,5)

**Comment by sulimanp** [comment]
The [libsdl2-dev](https://packages.ubuntu.com/focal/libsdl2-dev)  package must be installed
`sudo apt install libsdl2-dev`

**Comment by brunoais** [comment]
In my case, I can't install `libsdl2-dev` because it depends on libudev1=245.4-4ubuntu3.**4** but I have libudev1=245.4-4ubuntu3.**5**
Any alternative?

**Comment by rom1v** [comment]
```
sudo apt update
```
I guess.

**Comment by brunoais** [comment]
Nice try but I already tried all the obvious. I tried looking for different version, checking for other ppl asking for help... None of the straightforward is available. Seems to be a dependency of xubuntu-desktop, for some reason.

**Comment by rom1v** [comment]
I guess your apt is messed up. Is your `/etc/apt/sources.list` clean?

**Comment by brunoais** [comment]
No. I have there also packages from the manufacturer. This is an oficial image from them.
Besides that, I also have docker in there.
As for `/etc/apt/sources.list.d`, I 61 source files.
For `apt list --all-versions libudev1`
```
libudev1/now 245.4-4ubuntu3.5 amd64 [installed,local]
libudev1/focal-updates 245.4-4ubuntu3.4 amd64
libudev1/focal 245.4-4ubuntu3 amd64
```
